receiver/sqlqueryreceiver/config.go

```diff
@@ -4,186 +4,29 @@
-	ValueColumn      string            `mapstructure:"value_column"`
-	AttributeColumns []string          `mapstructure:"attribute_columns"`
-	Monotonic        bool              `mapstructure:"monotonic"`
-	ValueType        MetricValueType   `mapstructure:"value_type"`
-	DataType         MetricType        `mapstructure:"data_type"`
-	Aggregation      MetricAggregation `mapstructure:"aggregation"`
-	Unit             string            `mapstructure:"unit"`
-	Description      string            `mapstructure:"description"`
-	StaticAttributes map[string]string `mapstructure:"static_attributes"`
-	StartTsColumn    string            `mapstructure:"start_ts_column"`
-	TsColumn         string            `mapstructure:"ts_column"`
-}
-
-func (c MetricCfg) Validate() error {
-	var errs error
-	if c.MetricName == "" {
-		errs = multierr.Append(errs, errors.New("'metric_name' cannot be empty"))
-	}
-	if c.ValueColumn == "" {
-		errs = multierr.Append(errs, errors.New("'value_column' cannot be empty"))
-	}
-	if err := c.ValueType.Validate(); err != nil {
-		errs = multierr.Append(errs, err)
-	}
-	if err := c.DataType.Validate(); err != nil {
-		errs = multierr.Append(errs, err)
-	}
-	if err := c.Aggregation.Validate(); err != nil {
-		errs = multierr.Append(errs, err)
-	}
-	if c.DataType == MetricTypeGauge && c.Aggregation != "" {
-		errs = multierr.Append(errs, fmt.Errorf("aggregation=%s but data_type=%s does not support aggregation", c.Aggregation, c.DataType))
-	}
-	if errs != nil && c.MetricName != "" {
-		errs = multierr.Append(fmt.Errorf("invalid metric config with metric_name '%s'", c.MetricName), errs)
-	}
-	return errs
-}
-
-type MetricType string
-
-const (
-	MetricTypeUnspecified MetricType = ""
-	MetricTypeGauge       MetricType = "gauge"
-	MetricTypeSum         MetricType = "sum"
-)
-
-func (t MetricType) Validate() error {
-	switch t {
-	case MetricTypeUnspecified, MetricTypeGauge, MetricTypeSum:
-		return nil
-	}
-	return fmt.Errorf("metric config has unsupported data_type: '%s'", t)
-}
-
-type MetricValueType string
-
-const (
-	MetricValueTypeUnspecified MetricValueType = ""
-	MetricValueTypeInt         MetricValueType = "int"
-	MetricValueTypeDouble      MetricValueType = "double"
-)
-
-func (t MetricValueType) Validate() error {
-	switch t {
-	case MetricValueTypeUnspecified, MetricValueTypeInt, MetricValueTypeDouble:
-		return nil
-	}
-	return fmt.Errorf("metric config has unsupported value_type: '%s'", t)
-}
-
-type MetricAggregation string
-
-const (
-	MetricAggregationUnspecified MetricAggregation = ""
-	MetricAggregationCumulative  MetricAggregation = "cumulative"
-	MetricAggregationDelta       MetricAggregation = "delta"
-)
-
-func (a MetricAggregation) Validate() error {
-	switch a {
-	case MetricAggregationUnspecified, MetricAggregationCumulative, MetricAggregationDelta:
-		return nil
-	}
-	return fmt.Errorf("metric config has unsupported aggregation: '%s'", a)
+	return c.Config.Validate()
```

Comment:
I'd appreciate feedback on naming that would make sense here. `c.Config` is pretty clunky, but I'm not sure how best to name it since it's a config embedded within a config.